### ISSUE TITLE
Wikipedia bangs not working

### ISSUE THREAD

**Issue by cefe-origol** [hnhx/librex#185]
When trying to use the Wikipedia/Wikiless bangs(!w, !wen, !wiki, !wes, !whu), its not working. When searching for example "!w duck test", instead it redirects to "duck test?lang=en". I tried to fix it with my limited php skills, and got to this. I have no php interpreter in my computer, so i couldn't test it, but it should work as intended.
`/misc/tools.php`
```php
  42   │             if (strpos($url, "wikipedia.org") !== false)
  43   │             {
  44   │                 $wiki_split = explode(".", $url);
  45 ~ │                 if (count($wiki_split) > 1){
  46   │                      $lang = explode("://", $wiki_split[0])[1];
  47 ~ │                 }
  48 ~ │                 else{
  49 ~ │                     $lang = "en";
  50 ~ │                 }
  51 ~ │                 if (strpos($url, "Special:Search") !== false){
  52 ~ │                     # this is a search page
  53 ~ │                     $term = explode("&",explode("search=",$url)[1])[0];
  54 ~ │                     if ($term === ""){
  55 ~ │                     #if nothing was searched, go to the main page
  56 ~ │                         $url = $frontend . "/wiki/Main_Page?lang=" . $lang;
  57 ~ │                     }
  58 ~ │                     else{
  59 ~ │                         $url = $frontend . "/w/index.php?lang=en&search=" . $term . "&title=Special%3ASearch&profile=default&fulltext=1";
  60 ~ │                     }
  61 ~ │                 }
  62 ~ │                 else{
  63 ~ │                     # this is a regular page
  64 ~ │                     $url = $frontend . explode($original, $url)[1] . "?lang=" . $lang;
  65 ~ │                 }
```

**Comment by hnhx** [comment]
Thanks, I will fix in the next commit!

**Comment by hnhx** [comment]
Solved it in [35d9a21](https://github.com/hnhx/librex/commit/35d9a21c5d6c95d74ace9b3ea2ab647924d8db46)
